src/opm/output/eclipse/AggregateUDQData.cpp

```diff
@@ -64,30 +64,423 @@ namespace {
+                tmpPrec = opFuncPrec(expr.highestLevOperators[ind].type());
+                if (tmpPrec <= curPrec) {
+                    curPrec = tmpPrec;
+                    indLowestPrecOper = ind;
+                }
+            }
+        }
+        //
+        // if lowest precedence operator is the first token (and not equal to change sign)
+        // NOTE: also for the case with outer () removed
+        if ((!expr.leadChangeSign) && (indLowestPrecOper == 0)) {
+            // test if operator is a function (precedence = 6)
+            if ((curPrec == 6) || (expr.highestLevOperators[indLowestPrecOper].type() == Opm::UDQTokenType::binary_op_sub) ) {
+                def_type = -1;
+                def_type -= expr.noleadingOpenPar;
+            } else {
+                // def type is 1 when for all other situations (ecl-expression or number)
+                def_type = 1;
+            }
+        } else {
+            //
+            // treat cases which start either with (ecl_experessions, open-parenthes or leadChangeSign
+            def_type = (expr.leadChangeSign) ?  -1 : 0;
+            def_type -= expr.noleadingOpenPar;
+            // calculate position of lowest precedence operator
+            // account for leading change sign operator
+            for (std::size_t ind = 0; ind <= indLowestPrecOper; ind++) {
+                //
+                //count operators, including functions and parentheses (not original ecl_experessions)
+                if (operatorToken(expr.highestLevOperators[ind].type())) {
+                    // single operator - subtract one
+                    def_type -= 1;
+                } else if (expr.highestLevOperators[ind].type() == Opm::UDQTokenType::comp_expr) {
+                    // expression in parentheses -  add all operators
+                    std::size_t ind_ce = static_cast<std::size_t>(std::stoi(expr.highestLevOperators[ind].str()));
+                    auto indSubstTok = expr.substitutedTokens.find(ind_ce);
+                    if (indSubstTok != expr.substitutedTokens.end()) {
+                        auto& substTokVec = indSubstTok->second;
+                        //
+                        // count the number of operators & parenthes in this sub-expression
+                        def_type -= noOperators(substTokVec);
+                    } else {
+                        std::cout << "comp_expr index: " << ind_ce << std::endl;
+                        throw std::invalid_argument( "Invalid comp_expr index" );
+                    }
+                } else if ((expr.highestLevOperators[ind].type() != Opm::UDQTokenType::ecl_expr) &&
+                        (expr.highestLevOperators[ind].type() != Opm::UDQTokenType::number)) {
+                    // unknown token - write warning
+                    std::cout << "define_type - unknown tokenType: " << expr.highestLevOperators[ind].str() << std::endl;
```

Comment:
If it is a warning we should use `OpmLog::warning(....)` not `std::cout`

Reply:
The statement has been modified to a OpmLog::warning statement.